ocw/lib/decorators.py

```diff
@@ -0,0 +1,20 @@
+def filterAfterTime(*args, **kwargs):
+    def wrapper(func):
+        from datetime import datetime, timedelta
+        time_limit = datetime.now() - timedelta(hours=kwargs['hours'])
+
+        def time_filtered_list(*args, **kwargs):
+            dat = func(*args)
+            dt_frm = datetime.fromisoformat
+            return [k['name'] for k in dat if dt_frm(
+                    k['validAfterTime'][:-1]) < time_limit]
+        return time_filtered_list
+    return wrapper
+
+
+def filterService(*args, **kwargs):
```

Comment:
Maybe `filter_colllection_starts_with` and isn't it the same as https://docs.python.org/3/library/functions.html#filter

Reply:
i would say yes but from POV is just another way to choice of implementation. the decorator has different 'scope'. one is a design pattern, the other is python build-in function. So _same_ is relative